### PR TITLE
MNT: move piecewise functions to separate file

### LINKED ISSUE #667: MNT: Move the Piecewise function to a separate module
Reasons for solving this issue:
* The function.py file is already quite big (more than 3.5k code lines).
* We should define each class in a separate file, this makes the lib more readable.
* There is absolutely no need to define the Piecewise Function class in the same file as the `Function`class

Steps for accomplishing a solution to this issue:
- [x] Create a file named `rocketpy\mathutils\piecewise_function.py`
- [x] Move the `class PiecewiseFunction(Function):` definition to the new file
- [x] Updates the `mathutils/__init__.py` file to ensure the imports are working
- [x] Run the tests (`make pytest`) and verify everything is working properly
- [x] Beware of potential breaking changes. What is the behavior of the following command before and after the changes?

```python
from rocketpy.mathutils.function import PiecewiseFunction
```

### PR DESCRIPTION
**Pull request by yogirajgutte**
closes #667

## Pull request type

- [ ] Code changes (bugfix, features)
- [x] Code maintenance (refactoring, formatting, tests)
- [ ] ReadMe, Docs and GitHub updates
- [ ] Other (please describe):

## Checklist

- [x] Tests for the changes have been added (if needed)
- [x] Docs have been reviewed and added / updated
- [x] Lint (`black rocketpy/ tests/`) has passed locally 
- [x] All tests (`pytest tests -m slow --runslow`) have passed locally
- [ ] `CHANGELOG.md` has been updated (if relevant)

## Current behavior

Enter text here...

## New behavior

Enter text here...

## Breaking change

- [ ] Yes
- [x] No

## Additional information

Enter text here...

### CODE REVIEW

**Comment by Lucas-Prates** [comment]
@Gui-FernandesBR and @phmbressan, besides applying the code formaters and creating the tests, I simplified and optimized the `PiecewiseFunction` class. If you can, please take a very quick glance just to make sure it LGTY.

**Review by phmbressan — APPROVED** [review]
Before merging, the CHANGELOG should be updated.

File: rocketpy/mathutils/function.py
Comment (Gui-FernandesBR) [inline rocketpy/mathutils/function.py:1]: `from rocketpy.mathutils.function import PiecewiseFunction` will no longer work, so this is a breaking change.

We could import the class here, and then add a deprecation warning.

Comment (Gui-FernandesBR) [reply rocketpy/mathutils/function.py:1]: only concern is related to circular dependency import.

Or maybe we should just release with the change...

Comment (Lucas-Prates) [reply rocketpy/mathutils/function.py:1]: Your call!

Comment (Gui-FernandesBR) [reply rocketpy/mathutils/function.py:1]: I think we can live with that breaking change.
If anyone complains, we tell them the issue was opened for a while already.

**Review by Gui-FernandesBR — APPROVED** [review]
LGTM

**Pull request merged**
